orchestrator/modules/tools/execution/tool_grants.py

```
@@ -0,0 +1,434 @@
+
+        details = grant.details if isinstance(grant.details, dict) else {}
+        if details.get("params_hash") != params_hash(params):
+            # Params drifted from what the human approved — that is a new ask.
+            return None
+
+        stored_risk = grant.risk_tier or _risk_class_for(action, permission_level)
+        if stored_risk == RISK_DESTRUCTIVE:
+            # Single-use: the yes covered exactly one execution.
+            revoke_grant(grant, revoked_by=GRANT_CONSUMED_BY)
+        return grant
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)